### PR TITLE
Update Dokka & Dokkatoo versions

### PR DESCRIPTION
**Pull request by jbartok**
### Contributor Checklist
- [ ] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md).
- [ ] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [ ] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [ ] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team.
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective.
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic.
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes.
- [ ] Ensure that tests pass sanity check: `./gradlew sanityCheck`.
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -1430,9 +1432,9 @@
             <pgp value="33FD4BFD33554634053D73C0C2148900BCD3C2AF"/>
          </artifact>
       </component>
-      <component group="org.jetbrains" name="markdown-jvm" version="0.3.1">
-         <artifact name="markdown-jvm-0.3.1.jar">
-            <pgp value="E62231331BCA7E1F292C9B88C1B12A5D99C0729D"/>
+      <component group="org.jetbrains" name="markdown-jvm" version="0.5.2">
+         <artifact name="markdown-jvm-0.5.2.jar">
+            <sha256 value="726484477260a552dc7c19b09d4656c48ae3b10db2725000e182d6159529b46e" origin="Generated by Gradle" reason="A key couldn't be downloaded"/>
```
Comment (cobexer) [inline gradle/verification-metadata.xml:1437]: ❌ this must be resolved
> Also, there should not be any "Generated by Gradle" origins left.

**Review by cobexer — CHANGES_REQUESTED** [review]
Changes to the verification-metadata must be verified

**Comment by aSemy** [comment]
Hey @jbartok, I recommend you remove the Dokka version override here:

https://github.com/gradle/gradle/blob/4ed68e856d43f6b66e400e698e46265c43777937/platforms/documentation/docs/build.gradle#L101-L103

[Dokkatoo 2.3.1 uses Dokka 1.9.20 by default](https://github.com/adamko-dev/dokkatoo/blob/v2.3.1/gradle/libs.versions.toml#L4).

**Comment by jbartok** [comment]
> Hey @jbartok, I recommend you remove the Dokka version override ...

Very good point, thanks a lot!

**Comment by jbartok** [comment]
@cobexer good to go?

**Pull request merged**